src/shogun/kernel/GaussianKernel.h

```diff
@@ -174,6 +185,7 @@ class CGaussianKernel: public CShiftInvariantKernel
 protected:
 	/** width */
 	float64_t m_log_width;
+	Eigen::AutoDiffScalar<Eigen::VectorXd> m_eigen_log_width;
```

Comment:
this idea is actually not too bad: just make the parameters one can diff wrt class members. They just wrap the other member if I understand correctly? These two getting out of sync is he problem you mean with the user putting a parameter? Callback stuff might help? Should be possible to automate something for that or?

Reply:
a more consequent version of this would be to make all model parameters live in eigen, and all the "model" function simply returning composed versions of those